### DISCUSSION TITLE
[Feature Request] ClickHouse backend support

### DISCUSSION THREAD

**Discussion by marswong** [umami-software/umami#909] [Ideas]
Traditional RDBMS is really slow at large-scale OLAP, it's time to speed up OLAP with a better database and ClickHouse is a good choice today.

**Comment by mikecao** [comment]
We are limited to what is supported by Prisma. Adding any other database would take significant work.

**Comment by Form1ca** [comment]
> We are limited to what is supported by Prisma. Adding any other database would take significant work.


i was thinking about using clickhouse.
if you are interested i can add support clickhouse

Also have to use something for the cache - for example rabbitmq

**Comment by Kdecherf** [comment]
Should ClickHouse support be added to umami, please do not make it a requirement as I turned away Plausible because of that :pray:

**Comment by Form1ca** [comment]
> Should ClickHouse support be added to umami, please do not make it a requirement as I turned away Plausible because of that 🙏

what problem you get in Plausible and clickhouse?

**Comment by Kdecherf** [comment]
@Form1ca I'm using umami for a small blog on a server with restrained resources. When I tested Plausible, ClickHouse consumed too much resources for my needs, backup was not fun at all and in overall it was overkill for my workload. I don't find this stack relevant for monosite/small sites.

**Reply by Daniel15** [reply]
Did you modify the Clickhouse config to turn off logging? That's what caused a lot of the resource consumption in older versions. Plausible now ships with a default config to disable all the logging.

**Comment by entrptaher** [comment]
This is probably possible with clickhouse/postgres docker for fun projects. https://github.com/wavy/postgres-clickhouse

**Comment by Daniel15** [comment]
A column-based database like Clickhouse makes way more sense for analytics data compared to a row-based database, since practically all queries are aggregations of individual columns. You don't actually care about loading individual rows. OLAP-ish queries are much faster with a columnar database.

It doesn't *have* to be Clickhouse; that's just the most common one for this use case at the moment.

I agree with the other comments that it should be optional since very small deployments might not want to use it.
